core/src/main/java/io/confluent/kafka/schemaregistry/rest/resources/ConfigResource.java

```diff
@@ -111,12 +111,16 @@ public ConfigUpdateRequest updateSubjectLevelConfig(
   public Config getSubjectLevelConfig(@PathParam("subject") String subject) {
     Config config = null;
     try {
+      if(!schemaRegistry.hasSubjectConfig(subject)) {
+          throw Errors.subjectNotFoundException();
+      }
       AvroCompatibilityLevel compatibilityLevel = schemaRegistry.getCompatibilityLevel(subject);
       config = new Config(compatibilityLevel == null ? null : compatibilityLevel.name);
```

Comment:
If we're throwing the error above, we can ditch the null check in this constructor.